frontend/src/app/HashRouterElement.tsx

```diff
@@ -57,9 +73,14 @@ export const HashRouterElement = ({
     }
   }
 
-  return isAuthenticated ? (
-    element
-  ) : (
-    <Navigate replace to={redirectTo} state={{ from: location }} />
+  if (isAuthenticated && strict) {
+    return <Navigate to={state?.from.pathname ?? ROOT_ROUTE} replace />
+  }
+
+  return (
+    <Flex flexDir="column" height="100vh" pos="relative">
+      <GovtMasthead />
+      {element}
+    </Flex>
```

Comment:
/torefactor: Probably better to have `HashRouterElement` return a `PublicElement` rather than duplicate `PublicElement`'s code 🤔